optimade/server/entry_collections.py

```diff
@@ -119,19 +119,18 @@ def find(
             fields = set(params.response_fields.split(","))
         else:
             fields = all_fields.copy()
-        results = []
 
-        # build up set of included data with unique IDs
-        included = []
-        id_set = set()
+        results = []
+        included = {}
         for doc in self.collection.find(**criteria):
             results.append(self.resource_cls(**self.resource_mapper.map_back(doc)))
-            for reference in (
-                doc.get("relationships", {}).get("references", {}).get("data", [])
-            ):
-                if reference["id"] not in id_set:
-                    included.append(reference)
-                    id_set.add(reference["id"])
+            # collapse list of references into dict by ID and take only one per ID
+            refs = doc.get("relationships", {}).get("references", {}).get("data", [])
```

Comment:
For accuracy (since the db data may be stored in any which way) this should be:
```suggestion
            refs = results[-1].get("relationships", {}).get("references", {}).get("data", [])
```
And even then, it shouldn't explicitly be '"references"`. But for now it's fine.